### ISSUE TITLE
error TS2504: Type 'AsyncIterator<Plan, any, undefined>' must have a '[Symbol.asyncIterator]()' method that returns an async iterator.

### ISSUE THREAD

**Issue by anishkny** [recurly/recurly-client-node#73]
**Describe the bug**

Upgrading to recurly Node.js client v3.3.0 (released yesterday Dec 19) causes TypeScript v3.7.4 build failure with this code:

```ts
  for await (const plan of client.listPlans().each()) {
    plans.push(plan);
  }
```

```
error TS2504: Type 'AsyncIterator<Plan, any, undefined>' must have a '[Symbol.asyncIterator]()' method that returns an async iterator.
```

--

**To Reproduce**

Steps to reproduce the behavior.

If possible, please provide example code which we can run to reproduce the bug.
The faster we can reproduce the bug, the faster we can help you.

**Expected behavior**

**Your Environment**

- Which version of this library are you using? 3.3.0
- Which version of the language are you using? Node 12, TypeScript 3.7.4

**Comment by bhelx** [comment]
@anishkny I was able to repro this. We just released the TS definitions yesterday so there may be a few small issues. I'll prepare a fix. If you are urgently blocked, you could try 3.2.x.

**Comment by anishkny** [comment]
@bhelx thanks! Yes thats what I am doing right now (pinning to 3.2.x) - it may be worth investing in TS tests for basic flows (I see you have JS client tests and thats awesome!)

**Comment by bhelx** [comment]
Yes, great point @anishkny. When we fix this problem we will see how some testing can prevent future issues.

**Comment by bhelx** [comment]
@anishkny could you confirm that #74 fixes the problem? Seems to on my end.

**Comment by anishkny** [comment]
@bhelx yes! #74 seems to fix things on my end. Thanks for the fast turnaround! Let us know then its pushed. 🙏

**Comment by bhelx** [comment]
@anishkny this was released in [3.3.1](https://github.com/recurly/recurly-client-node/releases/tag/3.3.1)

I'm making a follow up ticket now for the typescript tests.
